src/stream/src/executor/source/source_executor.rs

```diff
@@ -245,27 +245,57 @@ impl<S: StateStore> SourceExecutor<S> {
             .stack_trace("source_recv_first_barrier")
             .await
             .unwrap();
+        let epoch = barrier.epoch.prev;
```

Comment:
Executor should use `epoch.cur` instead of `epoch.prev` for state store reads/writes after receiving a barrier. `epoch.cur` is the write epoch of data written after receiving a barrier and these data will be checkpointed on the next barrier. All data associated with `epoch.prev` will be checkpointed on seeing a barrier and `epoch.prev` can be INVALID_EPOCH if it comes from the first barrier our system sees.